build errors under MSVC

I'm revendoring this repo into https://github.com/oconnor663/kangarootwelve_xkcp.rs, and that's triggered a build failure on Windows under MSVC. See this CI run: https://github.com/oconnor663/kangarootwelve_xkcp.rs/actions/runs/3292387615/jobs/5427689462. The error is:

```
  running: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-I" "XKCP-K12/lib" "-I" "XKCP-K12/lib/Optimized64" "-W4" "/arch:AVX2" "-DLITTLE_ENDIAN=1" "-DKeccakP1600_noAssembly=1" "-FoD:\\a\\kangarootwelve_xkcp.rs\\kangarootwelve_xkcp.rs\\target\\debug\\build\\kangarootwelve_xkcp-6c3db5c94a44740e\\out\\XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.o" "-c" "XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.c"
  KeccakP-1600-timesN-AVX2.c
  XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.c(36): error C2054: expected '(' to follow 'rho8'
  XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.c(36): error C2143: syntax error: missing ';' before '='
  XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.c(37): error C2054: expected '(' to follow 'rho56'
  XKCP-K12/lib/Optimized64/KeccakP-1600-timesN-AVX2.c(37): error C2143: syntax error: missing ';' before '='
  exit code: 2
```

It looks like the trouble comes from [these alignment directives](https://github.com/XKCP/K12/blob/27a84ecb811200990add07a047214f4ee7f77d48/lib/Optimized64/KeccakP-1600-timesN-AVX2.c#L36-L37), which were introduced in dbfe81a3b05db1f54081ffce491e2783b2a74230. I've hacked in some `#error` lines to confirm that MSCV is taking [the appropriate branch in `align.h`](https://github.com/XKCP/K12/blob/27a84ecb811200990add07a047214f4ee7f77d48/lib/align.h#L26-L27), but it's still very possible I'm doing something weird on my end. I've reproduced this failure above on GitHub CI and also separately on a Windows laptop running MSVC 14.29.30133. Any chance this is a legit build break?

--- a/lib/Optimized64/KeccakP-1600-timesN-AVX2.c
+++ b/lib/Optimized64/KeccakP-1600-timesN-AVX2.c
@@ -33,8 +33,8 @@ Please refer to the XKCP for more details.
 #define ROL64in256(d, a, o)     d = _mm256_or_si256(_mm256_slli_epi64(a, o), _mm256_srli_epi64(a, 64-(o)))
 #define ROL64in256_8(d, a)      d = _mm256_shuffle_epi8(a, CONST256(rho8))
 #define ROL64in256_56(d, a)     d = _mm256_shuffle_epi8(a, CONST256(rho56))
-static const uint64_t rho8[4] ALIGN(32) = {0x0605040302010007, 0x0E0D0C0B0A09080F, 0x1615141312111017, 0x1E1D1C1B1A19181F};
-static const uint64_t rho56[4] ALIGN(32) = {0x0007060504030201, 0x080F0E0D0C0B0A09, 0x1017161514131211, 0x181F1E1D1C1B1A19};
+static ALIGN(AVX2alignment) const uint64_t rho8[4] = {0x0605040302010007, 0x0E0D0C0B0A09080F, 0x1615141312111017, 0x1E1D1C1B1A19181F};
+static ALIGN(AVX2alignment) const uint64_t rho56[4] = {0x0007060504030201, 0x080F0E0D0C0B0A09, 0x1017161514131211, 0x181F1E1D1C1B1A19};
 #define STORE256(a, b)          _mm256_store_si256((__m256i *)&(a), b)
 #define STORE256u(a, b)         _mm256_storeu_si256((__m256i *)&(a), b)
 #define XOR256(a, b)            _mm256_xor_si256(a, b)

